hop/models.py

```diff
@@ -148,41 +148,41 @@ def from_email(cls, email_input):
         )
 
     @classmethod
-    def from_email_file(cls, filename):
+    def load_file(cls, filename):
         """Create a new GCNCircular from an RFC 822 formatted circular file.
 
         Args:
-            filename: the GCN filename
+            filename: The GCN filename.
 
         Returns:
             The GCNCircular.
 
         """
-
         with open(filename, "r") as f:
-            return cls.from_email(f)
+            return cls.load(f)
 
 
 @dataclass
 class MessageBlob(object):
     """Defines an unformatted message structure.
 
-    This is included as a dataclass to mirror the implementation of structured formats.
+    This is included to mirror the implementation of structured formats.
 
     """
 
     content: str
+    missing_schema: bool = False
```

Comment:
FYI -- this shows up as a key/value whenever doing `subscribe ... --json` for `CIRCULAR` and `BLOB` formats, but not `VOEVENT`. Maybe we could ignore it if it's false, but this is a small detail for now.

Reply:
I went ahead with what you suggested, as in not show this attribute unless it's set to true. Running some tests with this, this only popped up in the `BLOB` format but not the `CIRCULAR` format. As in, if you publish a circular as a blob, it'll show up, but publishing it as a circular does not.

Reply:
That looks about right. Though with `subscribe`, I now don't see any reference to `missing_schema` (even for messages published with no specified format).

At first pass, I think this is because I can't trigger the [missing_schema line](https://github.com/scimma/hop-client/blob/update/adc_api/hop/io.py#L138) with either `subscribe` or `stream`, as any message that is lacking a specified format is "rescued" into a `BLOB` format, though I might be missing something.

(Though in general, I don't think this is necessarily bad behavior)

Reply:
Out of curiosity, how are you producing the poorly-formatted messages? The way it's set up, nothing you can send through hop will trigger missing_schema=True or raise the incorrectly formatted error.

The missing_schema line can be triggered by (1) extending the Deserializer to take in extra formats that a downstream subscriber doesn't have access to, or (2) sending in a message with the form {"format": "something", "content": "something"} which is JSON serialized and utf-8 encoded. Example:

```
[patrick@localhost ~/workspace/projects/github/scimma/hop-client] [update/adc_api] $ python
Python 3.7.4 (default, Jul  9 2019, 16:32:37) 
[GCC 9.1.1 20190503 (Red Hat 9.1.1-1)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import confluent_kafka, json
>>> p = confluent_kafka.Producer({"bootstrap.servers": "localhost:9092", "group.id": "hey"})
>>> msg = json.dumps({"format": "bad", "content":{"hey": "you"}}).encode("utf-8")
>>> p.produce("test3", msg)
>>> p.flush(0)
0
>>> 
[patrick@localhost ~/workspace/projects/github/scimma/hop-client] [update/adc_api] $ hop subscribe kafka://localhost:9092/test3 -s EARLIEST --no-auth --json
Message format BAD not recognized, returning a MessageBlob
{"content": {"hey": "you"}, "missing_schema": true}
```

The incorrectly-formatted error can be triggered in a similar way, except that the message format is something that the Consumer is not able to unpack.

Reply:
Got it. I wasn't extending or sending a JSON-formatted message with a bad format, just publishing a message that would be interpreted as `BLOB`. That all looks good to me now.